pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -127,10 +123,9 @@ func (d *namespacedResourcesDeleter) Delete(nsName string) error {
 		return nil
 	}
 
-	// Delete the namespace if it is already finalized.
-	if d.deleteNamespaceWhenDone && finalized(namespace) {
-		// TODO(liggitt): just return in 1.16, once n-1 apiservers automatically delete when finalizers are all removed
-		return d.deleteNamespace(namespace)
+	// return if it is already finalized.
+	if finalized(namespace) {
```

Comment:
For other reviewers: if the namespace is deleted by the previous updateStatus call, updateStatus returns the namespace object with empty finalizers (not a metav1.Status, as a regular Delete call would have returned), so this line is ok.